packages/toolkit/src/query/react/buildHooks.ts

```diff
@@ -93,11 +99,15 @@ export interface MutationHooks<
  * - Re-renders as the request status changes and data becomes available
  */
 export type UseQuery<D extends QueryDefinition<any, any, any, any>> = <
-  R extends Record<string, any> = UseQueryStateDefaultResult<D>
+  R extends Record<string, any> = UseQueryStateDefaultResult<D>,
+  Arg extends QueryArgFrom<D> | SkipToken = QueryArgFrom<D> | SkipToken
 >(
   arg: QueryArgFrom<D> | SkipToken,
   options?: UseQuerySubscriptionOptions & UseQueryStateOptions<D, R>
-) => UseQueryStateResult<D, R> & ReturnType<UseQuerySubscription<D>>
+) => UseQueryStateResult<D, R> &
+  ReturnType<UseQuerySubscription<D>> &
+  Suspendable &
+  IdleState<Arg>
```

Comment:
I'd suggest splitting this into an overloaded signature, one with `arg: SkipToken` and one without

Reply:
done!